synapse/types.py

```diff
@@ -852,25 +859,28 @@ def __bool__(self) -> bool:
         return bool(self.changed or self.left)
 
 
-def get_verify_key_from_cross_signing_key(key_info):
+def get_verify_key_from_cross_signing_key(
+    key_info: Dict[str, Any]
+) -> Tuple[str, VerifyKey]:
     """Get the key ID and signedjson verify key from a cross-signing key dict
 
     Args:
-        key_info (dict): a cross-signing key dict, which must have a "keys"
+        key_info: a cross-signing key dict, which must have a "keys"
             property that has exactly one item in it
 
     Returns:
-        (str, VerifyKey): the key ID and verify key for the cross-signing key
+        the key ID and verify key for the cross-signing key
     """
-    # make sure that exactly one key is provided
+    # make sure that a `keys` field is provided
     if "keys" not in key_info:
         raise ValueError("Invalid key")
     keys = key_info["keys"]
-    if len(keys) != 1:
-        raise ValueError("Invalid key")
-    # and return that one key
-    for key_id, key_data in keys.items():
+    # and that it contains exactly one key
+    if len(keys) == 1:
+        key_id, key_data = keys.popitem()
         return key_id, decode_verify_key_bytes(key_id, decode_base64(key_data))
+    else:
+        raise ValueError("Invalid key")
```

Comment:
This is meant to be equivalent. Mypy doesn't like the previous version: it can't see that the loop on -872 will always run exactly once.

Reply:
This appears to have broken a test!

Reply:
Ahh. If `keys` is empty, then before this change we would return `None`. Now we explicitly error.

Either the comments were incorrect, or else the implementation was buggy.

Reply:
This hasn't changed since its initial implementation! https://github.com/matrix-org/synapse/commit/c659b9f94fff29adfb2abe4f6b345710b65e8741#diff-d5643d4a94ee928bdd677a4ffbfbf3c571bc0a9f84d00d92dde31d91a6d53533R483-R501

Reply:
Every call site that I can see expects this function to return a 2-tuple.